### PR TITLE
Use data in modules, instead of params.pp

### PR DESCRIPTION
**Pull request by dhoppe**
#### Pull Request (PR) description
This is one of many pull requests:
- [x] Use data in modules, instead of params.pp 
- [ ] Use resource type archive, instead of curl / tar
- [ ] Use resource type systemd::unit_file, instead of file / exec resource
- [ ] Add support for Debian / Ubuntu, ...
- [ ] Refactor unit tests
- [ ] Refactor acceptance tests
- [ ] Add support for Puppet Strings
- [ ] ...

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -27,27 +27,36 @@
...
+  Optional[String[1]]            $caddy_account_id,
+  Optional[String[1]]            $caddy_api_key,
+  Enum['on','off']               $caddy_telemetry,
+  String                         $caddy_features,
+  Stdlib::Port                   $caddy_http_port,
+  Stdlib::Port                   $caddy_https_port,
+  Boolean                        $caddy_private_devices,
+  Integer                        $caddy_limit_processes,
+) {
+  case $::architecture {
```
Comment (ekohl) [inline manifests/init.pp]: `$facts['architecture']` perhaps?

File: spec/classes/caddy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+describe 'caddy', type: :class do
+  on_supported_os(facterversion: '3.11.7').each do |os, facts|
+    context "on #{os}" do
+      architecture = facts[:architecture]
+      os_family    = facts[:os]['family']
+      os_name      = facts[:os]['name']
+      # os_distro    = facts[:os]['distro']['codename']
+      os_release   = facts[:os]['release']['major']
+
+      case "#{os_name}-#{os_release}"
```
Comment (ekohl) [inline spec/classes/caddy_spec.rb]: IMHO it's cleaner to use a custom fact. https://github.com/theforeman/puppet-puppet/blob/c6d90aba578dd5ceb34b789f57e4bd7eac6a71cc/spec/spec_helper.rb#L18-L41 is an example.

File: manifests/init.pp (hunk 1 above)
Comment (dhoppe) [reply manifests/init.pp]: There are a lot of facts which require a rewrite.

Comment (ekohl) [reply manifests/init.pp]: We need a puppet-lint plugin that at least rewrites the built in ones.

**Comment by dhoppe** [comment]
> I think I dislike the common.yaml file for things that are the same everywhere. If you use static defaults in init.pp you can extract them with puppet-strings and get better docs.

I use ```common.yaml``` for some kind of default and additional files for each distro / release as required.

File: spec/classes/caddy_spec.rb (hunk 2 above)
Comment (dhoppe) [reply spec/classes/caddy_spec.rb]: I wanted to use the fact ```service_provider``` but it does not work as expected.

**Comment by bastelfreak** [comment]
@ekohl what do you think about this? Should data that's equal on all systems stay in the init.pp?

**Comment by ekohl** [comment]
@bastelfreak I do think so yes. It gives a better result in puppet-strings.

**Comment by bastelfreak** [comment]
I'm fine with that. I raised https://github.com/voxpupuli/voxpupuli.github.io/pull/189, lets see if it gets consent.

**Comment by b4ldr** [comment]
> I'm fine with that. I raised [voxpupuli/voxpupuli.github.io#189](https://github.com/voxpupuli/voxpupuli.github.io/pull/189), lets see if it gets consent.

just going through backlog, looks like this got consent now (and i agree)

**Comment by ekohl** [comment]
@dhoppe any intention to finish this or should we close this?

**Pull request closed without merging**